Subject: [patch] fix CC_FOR_BUILD problem
From: Robert Schwebel <r.schwebel@pengutronix.de>

Some tools for the "build" machine are by accident built for the "host"
box. Add CC_FOR_BUILD mechanics.

Description:
http://bugs.wireshark.org/bugzilla/show_bug.cgi?id=1185

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---

Index: Makefile.am
===================================================================
--- a/Makefile.am.orig
+++ b/Makefile.am
@@ -429,7 +429,7 @@ ps.c: print.ps rdps
 	./rdps $(srcdir)/print.ps ps.c
 
 rdps: rdps.c
-	$(CC) $(CFLAGS) -o rdps $(srcdir)/rdps.c
+	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) -o rdps $(srcdir)/rdps.c
 
 randpkt_additional_libs = wiretap/libwiretap.la
 
Index: tools/lemon/Makefile.am
===================================================================
--- a/tools/lemon/Makefile.am.orig
+++ b/tools/lemon/Makefile.am
@@ -34,8 +34,8 @@ CLEANFILES = \
 MAINTAINERCLEANFILES = \
 	Makefile.in
 
-lemon_SOURCES = \
-	lemon.c
+lemon: lemon.c
+	$(CC_FOR_BUILD) -D_U_="" $(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) -o $@ $<
 
 EXTRA_DIST = \
 	cppmagic.h		\
Index: configure.in
===================================================================
--- a/configure.in.orig
+++ b/configure.in
@@ -6,6 +6,7 @@ AC_PREREQ(2.52)
 
 dnl Check for CPU / vendor / OS
 AC_CANONICAL_HOST
+AC_CANONICAL_BUILD
 AC_CANONICAL_TARGET
 
 AM_INIT_AUTOMAKE(wireshark, 0.99.3a)
@@ -58,6 +59,12 @@ AC_SUBST(PYTHON)
 AC_SUBST(XSLTPROC)
 AC_SUBST(XMLLINT)
 
+if test "x$CC_FOR_BUILD" = x
+then
+       CC_FOR_BUILD=$(CC)
+fi
+AC_SUBST(CC_FOR_BUILD)
+
 # Check for doxygen
 AC_PATH_PROG(DOXYGEN, doxygen)
 AC_CHECK_PROG(HAVE_DOXYGEN, doxygen, "yes", "no")
